Src/FluentAssertions/Formatting/DefaultValueFormatter.cs

```diff
@@ -72,10 +80,44 @@ private static bool HasDefaultToStringImplementation(object value)
     private void WriteTypeAndMemberValues(object obj, FormattedObjectGraph formattedGraph, FormatChild formatChild)
     {
         Type type = obj.GetType();
-        formattedGraph.AddLine(TypeDisplayName(type));
-        formattedGraph.AddLine("{");
+        WriteTypeName(formattedGraph, type);
+        WriteTypeValue(obj, formattedGraph, formatChild, type);
+    }
+
+    private void WriteTypeName(FormattedObjectGraph formattedGraph, Type type)
+    {
+        var typeName = type.HasTypeName() ? TypeDisplayName(type) : string.Empty;
+        formattedGraph.AddFragment(typeName);
+    }
 
+    private void WriteTypeValue(object obj, FormattedObjectGraph formattedGraph, FormatChild formatChild, Type type)
+    {
         MemberInfo[] members = GetMembers(type);
+        if (members.Length == 0)
+        {
+            WriteEmptyType(formattedGraph);
+        }
+        else
+        {
+            WriteTypeValue(obj, formattedGraph, formatChild, members);
+        }
+    }
+
+    private static void WriteEmptyType(FormattedObjectGraph formattedGraph)
+    {
+        formattedGraph.AddFragment("{ }");
+    }
+
+    private static void WriteTypeValue(object obj, FormattedObjectGraph formattedGraph, FormatChild formatChild, MemberInfo[] members)
```

Comment:
🤔 I think the previous (main?) `WriteTypeValue` will be easier to understand if you would inline this and `WriteEmptyType` methods. This also avoids the need to call this one `WriteMemberValues` as well (because that's what it is doing)